File: frontend/src/api/agent.ts
import { apiClient } from './client';
import type { PipelineRunResponse } from '../types/api';

export interface ChatRequest {
  message: string;
}

export interface ChatResponse {
  assistant_message: string;
  tool_call?: {
    name: string;
    arguments: Record<string, any>;
  } | null;
  pipeline_result?: PipelineRunResponse | null;
}

export const sendChatMessage = async (message: string): Promise<ChatResponse> => {
  const response = await apiClient.post<ChatResponse>('/api/agent/chat', { message });
  return response.data;
};

File: dashboard_api/schemas/agent.py
from pydantic import BaseModel
from typing import Dict, Any, Optional
from dashboard_api.schemas.pipeline import PipelineRunResponse

class ChatRequest(BaseModel):
    message: str

class ChatResponse(BaseModel):
    assistant_message: str
    provider: str
    tool_call: Optional[Dict[str, Any]] = None
    pipeline_result: Optional[PipelineRunResponse] = None
